### PR TITLE
Add Dutch translations (Nederlands)

### PR DESCRIPTION
**Pull request by jeroenlammersma**
* Added Dutch translations

* Only 'minutes ago' and 'days ago' on lines [639](https://github.com/jeroenlammersma/openpilot/blob/851d8a9258f890629ffa1d176048c2d0e1d552d4/selfdrive/ui/translations/main_nl.ts#L639) and [649](https://github.com/jeroenlammersma/openpilot/blob/851d8a9258f890629ffa1d176048c2d0e1d552d4/selfdrive/ui/translations/main_nl.ts#L649) could be improved. In English you append an 's' if it's more than 1. However, in Dutch it's like this:

> minute = minuut
> minutes = minuten

> day = dag
> days = dagen

 
This is my solution to this issue:

**Minutes:**
```xml
<source>%1 minute%2 ago</source>
<translation>%1 min. geleden</translation>
```
**Days:**
```xml
<source>%1 day%2 ago</source>
<translation>%1 dag(en) geleden</translation>
```

But this solution is not optimal, it could be improved.

### CODE REVIEW

**Comment by sshane** [comment]
Thanks for the PR! Qt has a way to handle plurals in different languages, I just haven't gotten around to doing it yet. https://doc.qt.io/qt-6/i18n-source-translation.html

**Comment by jeroenlammersma** [comment]
You're welcome! More work than I initially anticipated, but I liked doing it.

If I have some spare time in the near future, I could have a look at handling plurals. But first, I need to finish my openpilot graduation project :)

In the meantime I thought about some improvements to the current translation and implemented them.

**Comment by sshane** [comment]
Could you check if I converted to plurals correctly?

**Comment by jeroenlammersma** [comment]
Yes you converted them correctly! 

Just wondering, does it also take into account 0 values? So for example::

> 0 minutes 
> 0 days

Will this translate to

> 0 minuten
> 0 dagen

also plural?

**Comment by sshane** [comment]
I know it does for English, yes

**Comment by jeroenlammersma** [comment]
Ok nice, then the translations will be correct for 0 values as well.

If you need more Dutch translations in the future, tag me or send me an email :)

**Comment by sshane** [comment]
Just wanted to check, seems like Dutch is a relatively popular language with openpilot after English, Mandarin, Japanese, Korean, Swedish, and German. And we already have half of those, so if you're willing to chip in here and there to make sure it's up to date before releases I'd be happy merging. Ideally and eventually more and more people can start contributing.

File: selfdrive/ui/translations/main_nl.ts [hunk 1]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="71"/>
+        <source>Show Map on Left Side of UI</source>
+        <translation>Toon kaart aan de linkerkant van de gebruikersinterface</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="72"/>
+        <source>Show map on left side when in split screen view.</source>
+        <translation>Toon kaart aan de linkerkant in gesplitste schermweergave.</translation>
```
Comment (sshane) [inline selfdrive/ui/translations/main_nl.ts:1225]: Do you know of any shorted translation? It pushes the toggles over a bit

File: selfdrive/ui/translations/main_nl.ts [hunk 2]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="100"/>
+        <source>N/A</source>
+        <translation>Nvt</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="101"/>
+        <source>Serial</source>
+        <translation>Serie</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:123]:
```suggestion
        <translation>Serienummer</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 3]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="113"/>
+        <source>Are you sure you want to reset calibration?</source>
+        <translation>Weet u zeker dat u de kalibratie wilt resetten?</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="120"/>
+        <source>Review Training Guide</source>
+        <translation>Doorloop de Training Opnieuw</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:158]:
```suggestion
        <translation>Doorloop de Training opnieuw</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 4]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="122"/>
+        <source>Are you sure you want to review the training guide?</source>
+        <translation>Weet u zeker dat u de training opnieuw wilt doorlopen?</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="130"/>
+        <source>Regulatory</source>
+        <translation>Regulatie</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:178]:
```suggestion
        <translation>Regelgeving</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 5]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="200"/>
+        <source>right</source>
+        <translation>rechts</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="211"/>
+        <source>Are you sure you want to reboot?</source>
+        <translation>Weet u zeker dat u het apparaat opnieuw wilt opstarten?</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:243]:
```suggestion
        <translation>Weet u zeker dat u opnieuw wilt opstarten?</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 6]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="218"/>
+        <source>Disengage to Reboot</source>
+        <translation>Deactiveer openpilot om het apparaat opnieuw op te starten</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:248]:
```suggestion
        <translation>Deactiveer openpilot om opnieuw op te starten</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 7]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="224"/>
+        <source>Are you sure you want to power off?</source>
+        <translation>Weet u zeker dat u het apparaat wilt uitschakelen?</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:253]:
```suggestion
        <translation>Weet u zeker dat u wilt uitschakelen?</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 8]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="231"/>
+        <source>Disengage to Power Off</source>
+        <translation>Deactiveer openpilot om het apparaat uit te schakelen</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:258]:
```suggestion
        <translation>Deactiveer openpilot om uit te schakelen</translation>
```

File: selfdrive/ui/translations/main_nl.ts (hunk 5 above)
Comment (pd0wm) [reply selfdrive/ui/translations/main_nl.ts:243]: Let's try to match the English as closely as possible.

(for other reviewers: the suggested translation means "Are you sure you want to reboot the device")

File: selfdrive/ui/translations/main_nl.ts [hunk 9]
```diff
@@ -0,0 +1,1305 @@
...
+        <source>Updating files: </source>
+        <translation>Bestanden bijwerken: </translation>
+    </message>
+</context>
+<context>
+    <name>MapETA</name>
+    <message>
+        <location filename="../qt/maps/map.cc" line="622"/>
+        <source>eta</source>
+        <translation>verwachte aankomsttijd</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:346]: This needs to be an abbreviation to fit in the bottom bar of the navigation. Let's just keep "eta".
```suggestion
        <translation>eta</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 10]
```diff
@@ -0,0 +1,1305 @@
...
+        <source>Get turn-by-turn directions displayed and more with a comma
+prime subscription. Sign up now: https://connect.comma.ai</source>
+        <translation>Krijg stapsgewijze routebeschrijving en meer met een comma
+prime abonnement. Meld u nu aan: https://connect.comma.ai</translation>
+    </message>
+    <message>
+        <location filename="../qt/maps/map_settings.cc" line="164"/>
+        <source>No home
+location set</source>
+        <translation>Geen thuislocatie ingesteld</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:425]:
```suggestion
        <translation>Geen thuislocatie
ingesteld</translation>
```

Comment (pd0wm) [reply selfdrive/ui/translations/main_nl.ts:425]: Not sure if this fits, needs testing

File: selfdrive/ui/translations/main_nl.ts [hunk 11]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/maps/map_settings.cc" line="169"/>
+        <source>No work
+location set</source>
+        <translation>Geen werklocatie ingesteld</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:431]:
```suggestion
        <translation>Geen werklocatie
ingesteld</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 12]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/maps/map_settings.cc" line="282"/>
+        <source>no recent destinations</source>
+        <translation>Geen recente bestemmingen</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:436]:
```suggestion
        <translation>geen recente bestemmingen</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 13]
```diff
@@ -0,0 +1,1305 @@
...
+        <location filename="../qt/onroad.cc" line="280"/>
+        <location filename="../qt/onroad.cc" line="283"/>
+        <source>MAX</source>
+        <translation>MAX</translation>
+    </message>
+    <message>
+        <location filename="../qt/onroad.cc" line="327"/>
+        <location filename="../qt/onroad.cc" line="330"/>
+        <source>SPEED</source>
+        <translation>SNELHEID</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:511]:
```suggestion
        <translation>SPEED</translation>
```
These represent the words "speed limit" in the US speed limit sign. They probably shouldn't be translated.

File: selfdrive/ui/translations/main_nl.ts [hunk 14]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/onroad.cc" line="333"/>
+        <location filename="../qt/onroad.cc" line="336"/>
+        <source>LIMIT</source>
+        <translation>LIMIET</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:517]:
```suggestion
        <translation>LIMIT</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 15]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/setup/setup.cc" line="248"/>
+        <source>Dashcam</source>
+        <translation>Dashcam</translation>
+    </message>
+    <message>
+        <location filename="../qt/setup/setup.cc" line="253"/>
+        <source>Custom Software</source>
+        <translation>Aangepaste Software</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:822]:
```suggestion
        <translation>Andere Software</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 16]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/setup/setup.cc" line="281"/>
+        <source>Enter URL</source>
+        <translation>Voer URL in</translation>
+    </message>
+    <message>
+        <location filename="../qt/setup/setup.cc" line="281"/>
+        <source>for Custom Software</source>
+        <translation>voor Aangepaste Software</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:832]:
```suggestion
        <translation>voor Andere Software</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 17]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/setup/setup.cc" line="303"/>
+        <source>Downloading...</source>
+        <translation>Downloaden...</translation>
+    </message>
+    <message>
+        <location filename="../qt/setup/setup.cc" line="315"/>
+        <source>Download Failed</source>
+        <translation>Download Mistlukt</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:842]:
```suggestion
        <translation>Download Mislukt</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 18]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/sidebar.h" line="41"/>
+        <source>3G</source>
+        <translation>3G</translation>
+    </message>
+    <message>
+        <location filename="../qt/sidebar.h" line="42"/>
+        <source>LTE</source>
+        <translation>LTE</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:977]:
```suggestion
        <translation>4G</translation>
```

File: selfdrive/ui/translations/main_nl.ts [hunk 19]
```diff
@@ -0,0 +1,1305 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="41"/>
+        <source>Receive alerts to steer back into the lane when your vehicle drifts over a detected lane line without a turn signal activated while driving over 31 mph (50 km/h).</source>
+        <translation>Ontvang waarschuwingen om terug naar de rijstrook te sturen wanneer uw voertuig over een gedetecteerde rijstrookstreep drijft zonder dat de richtingaanwijzer wordt geactiveerd terwijl u harder rijdt dan 50 km/u (31 mph).</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="46"/>
+        <source>Use Metric System</source>
+        <translation>Gebruik Metrische Systeem</translation>
```
Comment (pd0wm) [inline selfdrive/ui/translations/main_nl.ts:1180]:
```suggestion
        <translation>Gebruik Metrisch Systeem</translation>
```

**Comment by jeroenlammersma** [comment]
@sshane @pd0wm, maybe it's a good idea to add images of the UI, so you can easily see all source text 'in action'. Some texts really need context in order to translate them correctly. That could benefit the overall quality of a translation and may help potential future contributors.

What do you think?

File: selfdrive/ui/translations/main_nl.ts [hunk 20]
```diff
@@ -0,0 +1,1307 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="240"/>
+        <source>Last Update Check</source>
+        <translation>Laatste Updatecontrole</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="240"/>
+        <source>The last time openpilot successfully checked for an update. The updater only runs while the car is off.</source>
+        <translation>De laatste keer dat openpilot met succes heeft gecontroleerd op een update. De updater werkt alleen als de auto is uitgeschakeld.</translation>
```
Comment (sshane) [inline selfdrive/ui/translations/main_nl.ts:1017]: Is it possible to shorten this a bit? It gets cut off mid "updater"

File: selfdrive/ui/translations/main_nl.ts [hunk 21]
```diff
@@ -0,0 +1,1307 @@
...
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="253"/>
+        <source>ENTER</source>
+        <translation>INVOEREN</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="253"/>
+        <location filename="../qt/offroad/settings.cc" line="255"/>
+        <source>The new branch will be pulled the next time the updater runs.</source>
+        <translation>De nieuwe branch wordt opgehaald de volgende keer dat de updater wordt uitgevoerd.</translation>
```
Comment (sshane) [inline selfdrive/ui/translations/main_nl.ts:1043]: Is it possible to shorten this a bit? It gets cut off mid "updater"

File: selfdrive/ui/translations/main_nl.ts [hunk 22]
```diff
@@ -0,0 +1,1307 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="66"/>
+        <source>Use 24h format instead of am/pm</source>
+        <translation>Gebruik 24-uurs formaat in plaats van AM en PM</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="71"/>
+        <source>Show Map on Left Side of UI</source>
+        <translation>Toon kaart aan de linkerkant van de gebruikersinterface</translation>
```
Comment (sshane) [inline selfdrive/ui/translations/main_nl.ts:1222]: Is there a shorter word for `gebruikersinterface`? This makes the toggles pane horizontally scrollable

Comment (jeroenlammersma) [reply selfdrive/ui/translations/main_nl.ts]: Can you verify the 2 changes?

**Review by sshane — CHANGES_REQUESTED** [review]

**Comment by sshane** [comment]
Opened relevant issue: https://github.com/commaai/openpilot/issues/25492

File: selfdrive/ui/translations/main_nl.ts [hunk 23]
```diff
@@ -0,0 +1,1307 @@
...
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="241"/>
+        <source>Check for Update</source>
+        <translation>Controleer op Updates</translation>
+    </message>
+    <message>
+        <location filename="../qt/offroad/settings.cc" line="246"/>
+        <source>CHECKING</source>
+        <translation>CONTROLEREN</translation>
```
[the remark this answers is not in the document]
Comment (jeroenlammersma) [reply selfdrive/ui/translations/main_nl.ts:1027]: I changed it to 'CONTROLEER'. If that does not fit, you can also change it to: 'CHECKEN'. 'Checken' is a synonym for 'controleren', but 'controleren' fits best in this case.

**Review by jeroenlammersma — COMMENTED** [review]
Seems right.

**Review by jeroenlammersma — COMMENTED** [review]
Is outdated. Fixed it already by pushing an update. (Btw: I finally figured out how this 'change request' thing works on GitHub, so will do it the right way next time.)

**Comment by jeroenlammersma** [comment]
I ran update_translations.py but this changes nothing:
>  main_en.ts:
>      Found 233 source text(s) (229 new and 4 already existing)

>  rest:
>      Found 233 source text(s) (0 new and 233 already existing)

Do you know where the issue lies?

Btw, would update_translations.py be a useful pre-commit hook?

**Review by jeroenlammersma — COMMENTED** [review]
Outdated, but already fixed 'CONTROLEREN' (will do review changes request right next time, figured it out)

**Comment by jeroenlammersma** [comment]
Okay little bit lost now, I am new to reviewing changes on GitHub et cetera😅. What am I doing wrong exactly that the 'change requested' is still open? Sorry for the inconvenience btw.

**Review by sshane — APPROVED** [review]

File: selfdrive/ui/translations/languages.json [hunk 24]
```diff
@@ -1,5 +1,6 @@
 {
   "English": "main_en",
+  "Nederlands": "main_nl",
```
Comment (sshane) [inline selfdrive/ui/translations/languages.json:3]:
```suggestion
```

Comment (jeroenlammersma) [inline selfdrive/ui/translations/languages.json:3]: Do you mean the language or the country?
Because for language it should be 'Dutch'

**Comment by jeroenlammersma** [comment]
Nice!🙂

Like I said earlier, if you need new Dutch translations, you can send me a message or email!

**Pull request merged**